client/src/helpers/chatFunctions.tsx

```diff
@@ -125,30 +137,47 @@ export const joinChatFirebase = async (
   await updateDoc(doc(collections.chats, chatId), updates);
 };
 
-export const assignSupporter = async (userId: string) => {
+export const assignSupporter = async (userId: string, existingChatId?: string) => {
```

Comment:
I can see why it seems out of scope but it basically solves the following issue:
1. One person is waiting for a supporter
2. The supporter, when offline, is looking for a supportee
3. This will result in the supporter creating another searching room since he couldn't fetch any existing searches
4. So what this does is that if the supporter reloads/enters the chat again it will connect to the room that the supportee from step 1 has created